### ISSUE TITLE
Servo runs forever

### ISSUE THREAD

**Issue by shinglyu** [shinglyu/servo-perf#1]
Some tests in tp5 run forever, even if we set `-o output.png` and use `user-agent-js/02.timeout.js`. 
We need to investigate the root cause of this issue and decide we need to remove them from our test.

Currently failing tests:

http://localhost:8000/page_load_test/indiatimes.com/www.indiatimes.com/index.html
http://localhost:8000/page_load_test/media.photobucket.com/media.photobucket.com/image/funny%20gif/findstuff22/Best%20Images/Funny/funny-gif1.jpg@o=1.html
http://localhost:8000/page_load_test/people.com.cn/people.com.cn/index.html
http://localhost:8000/page_load_test/store.apple.com/store.apple.com/us@mco=Nzc1MjMwNA.html

**Comment by shinglyu** [comment]
@KuoE0 you might be interested in this.

**Comment by larsbergstrom** [comment]
@rzambre also had to disable many of the tests from the Alexa top sites due to this problem, IIRC, and may have a similar list.

**Comment by rzambre** [comment]
Yes, I had made a list of domain names for which Servo either took quite long to terminate (over 1 minute) or didn't terminate at all (within 2 minutes). I just tried out the domains in my earlier list using today's release version and it seems like some of the domains work well today :) I invoked Servo using the following: "time ./mach run -x -o out.png https://www.< domain name >"
Here, I'm reporting user times for the following domains:

- ehow.com: sometimes Servo terminates (less than 10 seconds), sometimes it doesn't (didn't terminate within 3 minutes)
- unisys.com: 2 min 34 s 
- michaels.com: didn't terminate within 3 min
- ebay.com: 2 min 55 s
- food.com: didn't terminate within 3 min
- indiatimes.com: didn't terminate within 3 min
- lifebuzz.com: didn't terminate within 3 min
- usmagazine.com: didn't terminate within 3 min
- today.com: 19 s
- slate.com: 35 s
- imation.com: 27 s

**Comment by larsbergstrom** [comment]
Interestingly, some of these (e.g., micheals.com) are still going after a solid 5 minutes, downloading more resources from the network repainting the page. I suspect there's some sort of madness going on here where Servo has pathologicially bad behavior for sites with really large numbers of images and script files being injected.

cc @jdm
